R/tags-diff.R

```diff
@@ -0,0 +1,127 @@
+#' Compare tags between models
+#'
+#' By default, this compares a model's tags to the tags on any of its "parent"
+#' models. A parent model is any model referenced in the relevant model's
+#' `based_on` field.  The `bbi_model` dispatch can optionally take a second
+#' model, in which case it will compare the first model to the second model,
+#' ignoring the `based_on` field entirely.
+#'
+#' @return
+#' **`tags_diff.bbi_model()`** invisibly returns a list with two
+#' elements, the first named `tags_added` and the second named `tags_removed`.
+#' The list is returned invisibly because, by default, a nicely formatted
+#' version of the same information is printed to the console. User can pass
+#' `.print = FALSE` to turn this off.
+#'
+#' **`tags_diff.bbi_run_log_df()`** returns a named list of lists, with one
+#' element for each row in the input tibble, with the name corresponding to the
+#' value in the `run` column for that row. Each element of the list will contain
+#' the two-element list returned from `tags_diff.bbi_model()` (described above)
+#' for the relevant model.
+#'
+#' **`add_tags_diff()`** returns the same tibble that was passed to
+#' it, but with two additional columns named `tags_added` and `tags_removed`.
+#'
+#' **In all cases:**
+#'
+#' * `tags_added` contains any tags that are on the relevant model, but _not_ on
+#'   any of the models it is based on.
+#'
+#' * `tags_removed` contains any tags that are on at least one of the models it
+#'   is based on, but _not_ on the relevant model.
+#'
+#' @param .bbi_object The object to compare. Could be a
+#'   `bbi_{.model_type}_model` object or a
+#'   tibble of class `bbi_run_log_df`.
+#' @param .mod2 If a `bbi_model` object is passed, compare `.bbi_object` to
+#'   `.mod2`. If `.mod2 = NULL`, the default, compare the `.bbi_object` model to
+#'   any models in its `based_on` field. **Only valid for `bbi_model`
+#'   method.**
```

Comment:
Add that `.mod2` is also a `bbi_{.model_type}_model` object?

Reply:
I have 
> If a `bbi_model` object is passed...
in there because this will work for Stan models, etc. too

Reply:
actually, you're right, it should be `bbi_{.model_type}_model`. I'll change it in [the `model_diff` PR](https://github.com/metrumresearchgroup/bbr/pull/341)